main/modes/test/keebTest/keebTest.c

```diff
@@ -0,0 +1,99 @@
+// Includes
+//==============================================================================
+
+#include "textEntry.h"
+
+// Function Prototypes
+//==============================================================================
+void keebEnterMode(void);
+void keebExitMode(void);
+void keebMainLoop(int64_t elapsedUs);
+
+// Variables
+//==============================================================================
+#define MAX_TEXT_LEN 32
+
+typedef struct
+{
+    wsg_t bg;
+    font_t fnt;
+    bool displayText;
+    char* typedText[MAX_TEXT_LEN];
+} keebTest_t;
+
+const char keebTestName[] = "Keyboard Test";
+
+swadgeMode_t keebTestMode = {
+    .modeName                 = keebTestName,
+    .wifiMode                 = NO_WIFI,
+    .overrideUsb              = false, 
+    .usesAccelerometer        = false,
+    .usesThermometer          = false,
+    .overrideSelectBtn        = false,
+    .fnEnterMode              = keebEnterMode,
+    .fnExitMode               = keebExitMode,
+    .fnMainLoop               = keebMainLoop,
+    .fnAudioCallback          = NULL,
+    .fnBackgroundDrawCallback = NULL,
+    .fnEspNowRecvCb           = NULL,
+    .fnEspNowSendCb           = NULL,
+    .fnAdvancedUSB            = NULL,
+};
+
+keebTest_t* kbTest;
+
+// Function definitions
+//==============================================================================
+
+void keebEnterMode(void)
+{
+    kbTest = (keebTest_t*)calloc(1, sizeof(keebTest_t));
+    loadWsg("menu_bg.wsg", &kbTest->bg, false);
+    loadFont("ibm_vga8.font", &kbTest->fnt, false);
+    kbTest->displayText = true;
+    strcpy(kbTest->typedText, "Press any key");
+}
+
+void keebExitMode(void)
+{
+    freeFont(&kbTest->fnt);
+    freeWsg(&kbTest->bg);
+    free(kbTest);
+}
+
+void keebMainLoop(int64_t elapsedUs)
+{
+    if (kbTest->displayText){
```

Comment:
Done in d73f5b3. @johnnywycliffe, you can resolve this comment

Reply:
I was individually running clang-format. Because my weird setup still doesn't work right, it wasn't finding it.

I'll work on it some more. Apologies.

Reply:
D'oh. the makefile _should_ be finding `clang-format` or `clang-format-17`. If it isn't that's a separate problem.

```make
# clang-format may actually be clang-format-17
CLANG_FORMAT:=clang-format
ifeq (, $(shell which $(CLANG_FORMAT)))
	CLANG_FORMAT:=clang-format-17
endif
```

Reply:
Nah, its because I have two separate makes with my cursed setup:
MingW64 can be used to make all, make clean, and run the swadge emulator.
powershell make can run make cppcheck, make format, etc. I didn't find this out until I fiddled around to get cppcheck to work, and then I forgot to go back for format...

Reply:
Are you running [Mingw-w64](https://www.mingw-w64.org/) and not [msys2](https://www.msys2.org/)?

Reply:
Er... No. MSYS2 is the install, I was just thinking of when I was setting the PATH variable.

Instead of opening up an MSYS2 terminal that's floating, I run the terminal from inside VSCode, though it's technically a "MSYS2 UCRT" window, not a straight up MSYS. They behave identically as far as I can tell, I've tried all the commands on both.